llvm/lib/Transforms/Utils/BuildLibCalls.cpp

```diff
@@ -66,6 +67,14 @@ static bool setIsCold(Function &F) {
   return true;
 }
 
+static bool setNoReturn(Function &F) {
+  if (F.hasFnAttribute(Attribute::NoReturn))
```

Comment:
Wouldn't it better to use `F.doesNotReturn()` and `setDoesNotReturn` as it is in `setDoesNotAccessMemory` instead of checking attributes explicitly? And rename to `setDoesNotReturn` probably...